homeassistant/components/diagnostics/__init__.py

```diff
@@ -213,12 +213,20 @@ async def get(
         sub_id: str | None = None,
     ) -> web.Response:
         """Download diagnostics."""
-        # t_type handling
+        # Validate d_type and sub_type
         try:
             d_type = DiagnosticsType(d_type)
```

Comment:
I think we should a different variable for the result here as the input is a different type than the result.